src/ucp/core/ucp_worker.c

```diff
@@ -1258,19 +1258,46 @@ ucp_worker_iface_add_distance(uct_iface_attr_t *attr,
     ucp_worker_iface_add_bandwidth(&attr->bandwidth, distance->bandwidth);
 }
 
+ucp_tl_md_t *ucp_worker_iface_get_md(const ucp_worker_iface_t *wiface)
+{
+    const ucp_context_h context       = wiface->worker->context;
+    const ucp_rsc_index_t rsc_index   = wiface->rsc_index;
+    const ucp_tl_resource_desc_t *rsc = &context->tl_rscs[rsc_index];
+    const ucp_md_index_t md_index     = rsc->md_index;
+
+    return &context->tl_mds[md_index];
+}
```

Comment:
There are multiple places with similar code in this file: ucp_worker_iface_get_sys_device, ucp_worker_init_device_atomics (the loop body). Please separate the common part into a separate function to avoid code duplication.